### PR TITLE
[SPARK-6124] Support jdbc connection properties in OPTIONS part of the query

### PR DESCRIPTION
**Pull request by vlyubin**
One more thing if this PR is considered to be OK - it might make sense to add extra .jdbc() API's that take Properties to SQLContext.

### CODE REVIEW

**Comment by aarondav** [comment]
Jenkins, add to whitelist

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala [hunk 1]
```diff
@@ -111,17 +113,20 @@ private[sql] class DefaultSource extends RelationProvider {
         numPartitions.toInt)
     }
     val parts = JDBCRelation.columnPartition(partitionInfo)
-    JDBCRelation(url, table, parts)(sqlContext)
+    val properties = new Properties() // Additional properties that we will pass to getConnection
```
Comment (aarondav) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: We should probably document this behavior somewhere -- I'm not sure if there's a standard place for data sources. If nowhere else, at least add it to the doc of createRelation().

**Comment by aarondav** [comment]
I think it would also be reasonable to add to the SQLContext#jdbc() methods a parameter for `options: Map[String, String]` like was done for createExternalTable. This would require confirmation by @marmbrus or @rxin, though.

**Comment by aarondav** [comment]
This LGTM, we'll merge this and later add a jdbc() version.

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala (hunk 1 above)
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: We should add this to the programming guide under JDBC.

Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: (i.e. http://spark.apache.org/docs/latest/sql-programming-guide.html#jdbc-to-other-databases)

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala [hunk 2]
```diff
@@ -115,18 +116,21 @@ private[sql] class DefaultSource extends RelationProvider {
...
+    JDBCRelation(url, table, parts, properties)(sqlContext)
   }
 }
 
 private[sql] case class JDBCRelation(
     url: String,
     table: String,
-    parts: Array[Partition])(@transient val sqlContext: SQLContext)
+    parts: Array[Partition],
+    properties: Properties = null)(@transient val sqlContext: SQLContext)
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: Why `null` as the default instead of an empty properties map?

**Comment by marmbrus** [comment]
One minor comment otherwise LGTM, provided documentation and a programatic interface are added in a follow-up PR.

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala (hunk 2 above)
Comment (vlyubin) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: No particular reason really, both are fine with DriverManagers' getConnection(). I've switched to empty properties map, I guess it is in fact neater than null.

**Comment by marmbrus** [comment]
Thanks!  Merged to master and branch-1.3.

**Pull request closed without merging**